test/test_webserver.py

```diff
@@ -24,13 +24,27 @@ def test_webserver():
     base_url = f"http://127.0.0.1:{http_port}"
     cluster_monitor_check_queue = Queue()
 
-    web = WebServer(config=config, cluster_state=cluster_state, cluster_monitor_check_queue=cluster_monitor_check_queue)
+    overall_state = {
+        "db_nodes": {},
+        "observer_nodes": {},
+        "current_master": "1",
+        "replication_slots_cache": {}
+    }
+
+    def get_overall_state_func():
+        return overall_state
+
+    web = WebServer(config=config, cluster_state=cluster_state, cluster_monitor_check_queue=cluster_monitor_check_queue,
+                    get_overall_state_func=get_overall_state_func)
     try:
         web.start()
         time.sleep(1)
         result = requests.get(f"{base_url}/state.json", timeout=5).json()
         assert result == cluster_state
 
+        result = requests.get(f"{base_url}/overall-state.json", timeout=5).json()
```

Comment:
I'm not a super fan of the state vs. overall-state names, but I guess we're stuck with it since state is directly a dict with the instances names at the root of the response :(

Reply:
yep, I am open for suggestions with better naming :)